### PR TITLE
Hunter 0.9.2

### PR DESCRIPTION
**Pull request by isaachier**
Second attempt.

### CODE REVIEW

File: lib/cpp/src/thrift/transport/TSSLSocket.cpp [hunk 1]
```diff
@@ -140,13 +145,11 @@ SSLContext::SSLContext(const SSLProtocol& protocol) {
   if(protocol == SSLTLS)
   {
     ctx_ = SSL_CTX_new(SSLv23_method());
-  }
-  else if(protocol == SSLv3)
-  {
+#ifndef OPENSSL_NO_SSL3
```
Comment (ruslo) [inline lib/cpp/src/thrift/transport/TSSLSocket.cpp:147]: Please use separate commit for this change. I guess it's not related to CMake.

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,121 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
```
Comment (ruslo) [inline CMakeLists.txt:2]: Is it cherry-picked from `hunter-0.10.0` branch? Can you add a note about it? Is there any difference?

**Review by ruslo — CHANGES_REQUESTED** [review]

File: lib/cpp/src/thrift/transport/TSSLSocket.cpp (hunk 1 above)
Comment (isaachier) [reply lib/cpp/src/thrift/transport/TSSLSocket.cpp:147]: I guess not but it does relate to default version of OpenSSL in Hunter. Seems to be older versions don't need this fix according to this pull request: https://github.com/apache/thrift/pull/1102.

File: CMakeLists.txt (hunk 2 above)
Comment (isaachier) [reply CMakeLists.txt:2]: CMake build seems to be new. Essentially checked out all CMake stuff from 0.10.0, then changed source files to match 0.9.2. Where should I add the note?

Comment (ruslo) [reply CMakeLists.txt:2]: > Essentially checked out all CMake stuff from 0.10.0

Commit 1: "CMake stuff from hunter-0.10.0 branch"

> then changed source files to match 0.9.2

Commit 2 (I guess it's https://github.com/hunter-packages/thrift/pull/3/commits/1d24cc93a10a721a27de8f88fcb2933ff9818ad9 already?)

File: lib/cpp/src/thrift/transport/TSSLSocket.cpp (hunk 1 above)
Comment (ruslo) [reply lib/cpp/src/thrift/transport/TSSLSocket.cpp:147]: > I guess not but it does relate to default version of OpenSSL in Hunter

Not sure I follow. It's not about CMake or Hunter. This change is about C++ code. Make a separate commit with this change.

> according to this pull request: apache#1102

Please add this link to commit too. Effectively it means we have two commits here.

**Review by ruslo — CHANGES_REQUESTED** [review]

**Comment by isaachier** [comment]
OK should be fixed now.

File: lib/cpp/src/thrift/transport/TSSLSocket.cpp (hunk 1 above)
Comment (ruslo) [inline lib/cpp/src/thrift/transport/TSSLSocket.cpp:147]: There is no such macro in https://github.com/apache/thrift/pull/1102/files change.

**Review by ruslo — CHANGES_REQUESTED** [review]

File: lib/cpp/src/thrift/transport/TSSLSocket.cpp (hunk 1 above)
Comment (isaachier) [reply lib/cpp/src/thrift/transport/TSSLSocket.cpp:147]: Oh wrong commit. Hold on...

**Review by ruslo — CHANGES_REQUESTED** [review]
I'm expecting two separate commits. One for https://github.com/apache/thrift/pull/1102/files change and one for https://github.com/apache/thrift/pull/944/files

**Comment by isaachier** [comment]
Done

**Review by ruslo — APPROVED** [review]

**Comment by ruslo** [comment]
Released: https://github.com/hunter-packages/thrift/releases/tag/v0.9.2-p0

**Pull request merged**